src/Core/Components/DateTime/CalendarExtended.cs

```diff
@@ -41,46 +41,20 @@ public IEnumerable<DateTime> GetDaysOfWeek(int weekNumber, int monthOffset = 0)
             throw new ArgumentException("Index must be between 0 and 5");
         }
 
-        var monthFirst = GetFirstDayToDisplay(monthOffset);
-        var maxLimit = monthFirst.Year == DateTime.MaxValue.Year && monthFirst.Month == DateTime.MaxValue.Month && weekNumber > 3;
+        var monthFirst = Date.AddMonths(monthOffset, Culture);
+        var maxDate = Culture.Calendar.MaxSupportedDateTime;
+        var maxLimit = monthFirst.Year == maxDate.Year && monthFirst.Month == maxDate.Month && weekNumber > 3;
 
         if (!maxLimit)
         {
-            var weekFirst = monthFirst.AddDays(weekNumber * 7).StartOfWeek(GetFirstDayOfWeek());
+            var weekFirst = monthFirst.AddDays(weekNumber * 7, Culture).StartOfWeek(Culture);
             for (var i = 0; i < 7; i++)
             {
-                yield return weekFirst.AddDays(i);
+                yield return weekFirst.AddDays(i, Culture);
             }
         }
     }
 
-    /// <summary>
-    /// Gets the first day of the n-th month.
-    /// </summary>
-    /// <returns></returns>
-    public DateTime GetFirstDayToDisplay(int monthOffset = 0)
```

Comment:
You cannot remove these `public` methods which are probably used by other developers.

Reply:
Although the method is `public`, the containing class is `internal`.
And the method is not used anywhere in the project.
This seems to be redundant.